### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**

### CODE REVIEW

**Review by rash-tiwari — APPROVED** [review]

File: SCF/Session 1/Assignment1/Item.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
+public class Item {
+	
+		private final String Name;
+		private int quantity=0;
+		private double Price;
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/Item.java]: Done

File: SCF/Session 1/Assignment1/Item.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+public class Item {
+	
+		private final String Name;
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/Item.java]: Done

File: SCF/Session 1/Assignment1/Item.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		
+		public int getQuantity(){
+			return this.quantity;
+		}
+		
+		public double getPrice(){
+			return this.Price;
+		}
+		
+		public void IncQuantity(int incQuantity){
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/Item.java]: Done

File: SCF/Session 1/Assignment1/Item.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+
+public class Item {
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/Item.java:2]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
+import java.util.*;
+
+public class ShoppingCart {
+	//cart store all the item which user will buy!
+	Set<Item> cart=new HashSet<Item>();
+	
+	//Hash map created in it key will be name of item and value will be object of that item! 
+	static Map< String, Item> items =  new HashMap< String, Item>();
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:9]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    
+	//initializing store items add them into HashMap items;
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:11]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 7]
```diff
@@ -0,0 +1,211 @@
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:1]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+		 	
+		 	//creating object of ShoopingCart Class for user! 
+		 	ShoppingCart cart = new ShoppingCart();
+			
+			//display available items with their price!
+			System.out.println("Available Items!");
+			System.out.println("Name \t\t price ");
+			
+			for (Map.Entry<String, Item> entry : items.entrySet()){  
+	            System.out.println(entry.getKey() + " \t\t " + entry.getValue().getPrice());
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:55]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+			
+			//creating a scanner object for taking inputs!
+		 	Scanner input = new Scanner(System.in);
+		 	
+		 	//creating object of ShoopingCart Class for user! 
+		 	ShoppingCart cart = new ShoppingCart();
+			
+			//display available items with their price!
+			System.out.println("Available Items!");
+			System.out.println("Name \t\t price ");
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+	        } 
+			
+			int choice;// taking choice from user!
+			
+			do{
+				System.out.println("Choose your Choice!");				
+				System.out.println("Enter 1 for Display items of Cart!");
+				System.out.println("Enter 2 for Add an item from Cart");
+				System.out.println("Enter 3 for Remove an item in cart!");
+				System.out.println("Enter 4 for when you are done and want to Generate bill!");
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:66]: Done

File: SCF/Session 1/Assignment1/ShoppingCart.java [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+		Item j = new Item("j",20.0);
+		items.put(j.getName(), j);
+		Item k = new Item("k", 60.5);
+		items.put(k.getName(), k);
+		Item l = new Item("l",34.0);
+		items.put(l.getName(), l);
+		
+	}
+	
+
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Session 1/Assignment1/ShoppingCart.java:41]: Done

**Pull request merged**